OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo

```diff
@@ -192,6 +194,79 @@ algorithm
   end if;
 end removeVerySimpleEquations;
 
+public function fixAliasVarsVariablity
+" This is a workaround for #5673
+  This module traverses all alias variables and double-checks if the referenced alias variables variablitly
+  is parameter or not and changes variablilty of alias variable to BackendDAE.PARAM()."
+  input BackendDAE.BackendDAE inDAE;
+  output BackendDAE.BackendDAE outDAE;
+protected
+  BackendDAE.Variables aliasVars,systvars,globalKnownVars;
+  DAE.Exp binding;
+  DAE.ComponentRef cr;
+  Boolean param;
+  BackendDAE.Shared shared;
+  list<BackendDAE.Var> referencevar,aliasVarList = {},knownVarList={};
+algorithm
+  aliasVars := BackendDAEUtil.getAliasVars(inDAE);
+  systvars := getVarsFromEqSystem(inDAE.eqs);
+  //BackendDump.dumpVariables(aliasVars,"aliasVariable-Actual");
+  for var in BackendVariable.varList(aliasVars) loop
+    binding := BackendVariable.varBindExp(var);
+    {cr} := Expression.getAllCrefs(binding);
+    // look up for the reference variable variablity
+    if BackendDAEUtil.containAnyVar({cr}, systvars) then  //check in EqSystem.OrderedVars
+      (referencevar,_) := BackendVariable.getVar(cr,systvars);
+    elseif BackendDAEUtil.containAnyVar({cr}, inDAE.shared.globalKnownVars) then //check in shared.globalKnownVars
+      (referencevar,_) := BackendVariable.getVar(cr,inDAE.shared.globalKnownVars);
+    else
+      // some other cases, example variables in shared.externalObjects, and shared.localKnownVars do not handle this case for now
+      referencevar :={};
+    end if;
+    param := checkVariability(referencevar);
+    if (param) then
+      // remove the variable from AliasVarList and add it to GlobalknownVarlist
+      knownVarList := BackendVariable.setVarKind(var,BackendDAE.PARAM()) :: knownVarList;
+      knownVarList := BackendVariable.setVarFixed(var,true) :: knownVarList;
```

Comment:
Why isn't this needed for alias variables?

Reply:
if we make it fixed = true then the varKind of alias variable is not changed if they are in aliasVarList

Reply:
❓

Reply:
Shouldn't the fixed attribute only affect states?

Reply:
I think i should not change fixed=true

Reply:
@lochel I should not change it i fixed=true, because with the latest update of omc i am getting some  kind of warning which i did not get it until yesterday